Microsoft.Toolkit.Uwp.UI.Controls.Graph/PeoplePicker/PeoplePicker.Events.cs

```diff
@@ -0,0 +1,159 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Graph;
+using Microsoft.Toolkit.Services.MicrosoftGraph;
+using Microsoft.Toolkit.Uwp.UI.Extensions;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls.Graph
+{
+    /// <summary>
+    /// Defines the events for the <see cref="PeoplePicker"/> control.
+    /// </summary>
+    public partial class PeoplePicker : Control
+    {
+        private static void AllowMultiplePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var control = d as PeoplePicker;
+            if (!control.AllowMultiple)
+            {
+                control.Selections.Clear();
+                control.RaiseSelectionChanged();
+                control._searchBox.Text = string.Empty;
+            }
+        }
+
+        private void ClearAndHideSearchResultListBox()
+        {
+            SearchResultList.Clear();
+            _searchResultListBox.Visibility = Visibility.Collapsed;
+        }
+
+        private async void SearchBox_OnTextChanged(object sender, TextChangedEventArgs e)
+        {
+            var textboxSender = (TextBox)sender;
+            string searchText = textboxSender.Text.Trim();
+            if (string.IsNullOrWhiteSpace(searchText))
+            {
+                ClearAndHideSearchResultListBox();
+                return;
+            }
+
+            _loading.IsActive = true;
```

Comment:
@michael-hawker IsLoading and AllowMultiple are done, however, the binding of `CollectionVisibilityConverter` (see below) doesn't work in our scenario, as the changes in this `ObservableCollection` couldn't trigger the visibility binding properly (RelativeSource of TemplatedParent is not necessary in our scenario because of the `DataContext` on the parent element), so we would continue to update the `Visibility` property directly for this control for now.

```
Visibility="{Binding SearchResultList, Converter={StaticResource CollectionVisibilityConverter}}"
```